lib/pos_blockchain.go

```diff
@@ -472,6 +488,48 @@ func (bc *Blockchain) GetUncommittedTipView() (*UtxoView, error) {
 	return utxoView, nil
 }
 
+// BuildUtxoViewToBlockParent builds a UtxoView to the block provided. It does this by
+// identifying all uncommitted ancestors of this block and then connecting those blocks.
+func (bc *Blockchain) BuildUtxoViewToBlockParent(parentHash BlockHash) (*UtxoView, error) {
```

Comment:
nit: this function shouldn't internally care about why it's being called. If we define this function as simple building a UtxoView given a blockHash, then it can become a general purpose function that we can reuse. Ex: you can reuse it to do the heavy lifting for `bc.getUncommittedTipView()`
```suggestion
func (bc *Blockchain) buildUtxoViewForBlock(blockHash BlockHash) (*UtxoView, error) {
```